### PR TITLE
docs: how to write snapshot tests

### PR DESCRIPTION
**Pull request by emmettbutler**
This change adds fairly detailed instructions on how to contribute integration tests to this repository.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: docs/contributing-integrations.rst [hunk 1]
```diff
@@ -179,6 +174,71 @@ to start the APM test agent, and re-run the test. Use `--pass-env` as described
...
+4. In this directory, write a simple "Hello World" application that uses the library you're
+   integrating with similarly to how customers will use it. Depending on the library, this
+   might be as simple as a function in the snapshot test file that imports the library.
+   It might also be a new file in the test directory ``app.py`` as in the cases of Flask
+   or Gunicorn.
+5. Instrument your "hello world" app with ddtrace. In the case of Flask, this is accomplished by
+   running the app server in a subprocess started with a ``ddtrace-run`` command. The app
+   server is started by a Pytest fixture function that's defined in the snapshot test file.
+6. If the library you're integrating with requires communication with a datastore, make sure there's
+   an image for that datastore referenced in ``doker-compose.yml``. If there is not, add one.
```
Comment (duncanista) [inline docs/contributing-integrations.rst:197]:
```suggestion
   an image for that datastore referenced in ``docker-compose.yml``. If there is not, add one.
```

File: docs/contributing-integrations.rst [hunk 2]
```diff
@@ -179,6 +174,71 @@ to start the APM test agent, and re-run the test. Use `--pass-env` as described
...
+   You can find a suitable image by searching on hub.docker.com.
```
Comment (duncanista) [inline docs/contributing-integrations.rst:198]: Should we add a link for this or not?

File: docs/contributing-integrations.rst [hunk 3]
```diff
@@ -179,6 +174,71 @@ to start the APM test agent, and re-run the test. Use `--pass-env` as described
...
+
+    @pytest.mark.snapshot
+    def test_flask_200(flask_client):
+        assert flask_client.get("/", headers=DEFAULT_HEADERS).status_code == 200
+
+
+This function accepts a client object, defined elsewhere in the file, as a fixture. The
+client has been initialized to communicate with the server running the "hello world" app we
+created in step 3. The function makes a simple request to the app server and checks the status
+code. The ``pytest.mark.snapshot`` decorator
```
Comment (duncanista) [inline docs/contributing-integrations.rst:212]:
```suggestion
code. The ``pytest.mark.snapshot`` decorator.
```

File: docs/contributing-integrations.rst [hunk 4]
```diff
@@ -179,6 +174,71 @@ to start the APM test agent, and re-run the test. Use `--pass-env` as described
...
+        ...
+
+
+This decorator causes Pytest to collect the spans generated by your instrumented test app and compare them
+against a stored set of expected spans. Since the integration test we're writing is new, there
+are not yet any expected spans stored for it, so we need to create some.
+
+9. Start any necessary datastore containers with ``docker-compose up -d <container>``. Run your new test with
+   ``scripts/ddtest riot -v run <test_suite_name>``.
+10. Check ``git status`` and observe that some new files have been created under ``tests/snapshots/``.
```
Comment (duncanista) [inline docs/contributing-integrations.rst:229]: Nice, really useful, normally I would try to find them on the directory.

**Review by duncanista — APPROVED** [review]
Couple nits, but it looks really good!

File: docs/contributing-integrations.rst [hunk 5]
```diff
@@ -179,6 +174,78 @@ to start the APM test agent, and re-run the test. Use `--pass-env` as described
...
+Writing Integration Tests for Your Integration
+++++++++++++++++++++++++++++++++++++++++++++++
+
+These instructions describe the general approach of writing new integration tests for a library integration.
+They use the Flask integration tests as a teaching example. Referencing these instructions against
+``tests/contrib/flask/test_flask_snapshot.py`` and ``tests/contrib/flask/app.py`` may be helpful.
+
+1. Make sure a directory for your integration exists under ``tests/contrib``
+2. Create a new file ``tests/contrib/<integration>/test_<integration>_snapshot.py``
+3. Make sure a ``Venv`` instance exists in ``Riotfile.py`` that references your ``contrib`` subdirectory.
```
Comment (brettlangdon) [inline docs/contributing-integrations.rst:186]:
```suggestion
3. Make sure a ``Venv`` instance exists in ``riotfile.py`` that references your ``contrib`` subdirectory.
```

File: docs/contributing-integrations.rst [hunk 6]
```diff
@@ -179,6 +174,78 @@ to start the APM test agent, and re-run the test. Use `--pass-env` as described
...
+   Create one if it doesn't exist. Note the name of this ``Venv`` - this is the "test suite name".
```
Comment (brettlangdon) [inline docs/contributing-integrations.rst:187]: should we provide a copy/paste-able example?

**Review by brettlangdon — APPROVED** [review]
little nits, but high level great starting place to get feedback from partners on!

File: docs/contributing-integrations.rst (hunk 6 above)
Comment (emmettbutler) [reply docs/contributing-integrations.rst:187]: Sure, I'll add that in a future change

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**